packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -23,13 +23,18 @@ class MDnsObservatoryDiscovery {
   /// Dart observatory ports.
   MDnsObservatoryDiscovery({
     MDnsClient? mdnsClient,
+    MDnsClient? preliminaryMDnsClient,
     required Logger logger,
     required Usage flutterUsage,
-  }): _client = mdnsClient ?? MDnsClient(),
-      _logger = logger,
-      _flutterUsage = flutterUsage;
+  })  : _client = mdnsClient ?? MDnsClient(),
+        _preliminaryClient = preliminaryMDnsClient,
+        _logger = logger,
+        _flutterUsage = flutterUsage;
 
   final MDnsClient _client;
+
+  final MDnsClient? _preliminaryClient;
```

Comment:
I see the comment in `queryForAttach` but this could use a comment up here to explain what it's for.